src/lib/storage/lz4_segment/lz4_segment_iterable.hpp

```diff
@@ -128,29 +136,31 @@ class LZ4SegmentIterable : public PointAccessibleSegmentIterable<LZ4SegmentItera
    public:
     using ValueType = T;
     using IterableType = LZ4SegmentIterable<T>;
+    using DataIteratorType = typename std::vector<T>::const_iterator;
+    using NullValueIterator = typename pmr_vector<bool>::const_iterator;
 
     // Begin Iterator
-    PointAccessIterator(const std::vector<T>& data, const std::optional<pmr_vector<bool>>* null_values,
-                        const PosList::const_iterator position_filter_begin, PosList::const_iterator position_filter_it)
+    PointAccessIterator(DataIteratorType data_it, const std::optional<NullValueIterator>& null_value_it,
```

Comment:
same: just `std::optional<NullValueIterator>`

Reply:
What? I removed that 5 times already.